### DISCUSSION TITLE
Is there a facility for dumping the contents of a NSKeyedArchive file

### DISCUSSION THREAD

**Discussion by ohinds** [danielpaulus/go-ios#146] [Q&A]
I have an NSKeyedArchive (CoreData) file writen by an iOS app locally and I'd like to inspect the contents. I see the nskeyedarchiver project has been folded into this one, is there a way to dump an unarchived version of the file using go-ios?

**Comment by danielpaulus** [comment]
hmm not yet. I can create a nskeyedunarchiver if you like. should be pretty straight forward. can you publish the archive so I can test it?

**Comment by nanoscopic** [comment]
I've additionally written NSKeyedArchive unarchiving code, but in C not in Go. It is part of the iosif project. I have not made it accessible via a easy CLI standalone option though. Same as Daniel is saying, it shouldn't be hard to pass data to it and get it unarchived.

There is also a key difference between my unarchiving code and all other libs I've seen that do it. I don't actually use the index/offset information of the format. I do a prescan of the data and then a second pass full decode. This lets one unarchive data with corrupt or missing index/offset info.

**Comment by nanoscopic** [comment]
Is this still something desired? @ohinds Could you explain how to fetch such a file so I can test a tool to do so??
